server/s3_url_encode.cc

```diff
@@ -73,3 +73,39 @@ std::string url_encode(const char* src) {
   }
   return encoded_string;
 }
+
+void remove_escape_char(const char* str, std::string& destination) {
+  char buf[3] = "\0";
+  unsigned int code = 0;
+  snprintf(buf, sizeof(buf), "%s", str + 1);
+  sscanf(buf, "%x", &code);
+  destination.append(1, (char)code);
+}
+
+bool is_url_encoded_char(char c) {
+  if (c == '%') {
+    return true;
+  }
+  return false;
+}
+
+std::string url_decode(const char* src) {
```

Comment:
please write UTs for copy_object where URL has encoded chars

Reply:
This currently can only be simulated via UI (like cloudberry). s3cmd CLI is not allowing for spaces in object file, but on Windows setup, we can created objects with spaces in name.

Reply:
in UT, we do not need to run any client or server... create mock URLs and test.